samples/client/petstore/typescript-angular2/default/api/pet.service.ts

```diff
@@ -117,7 +118,7 @@ export class PetService {
                 if (response.status === 204) {
                     return undefined;
                 } else {
-                    return response.json() || {};
+                    return ObjectSerializer.deserialize(response.json(), "Array&lt;Pet&gt;") || {};
```

Comment:
PLease use `{{{returnType}}}` instead to avoid HTML encoding performed by mustache.